File: P/1010-pairs-of-songs-with-total-durations-divisible-by-60/main.go
package main

import "fmt"

func numPairsDivisibleBy60(time []int) int {
	buf := make(map[int]int)
	for i := 0; i < len(time); i++ {
		buf[time[i]%60]++
	}

	c := 0
	for i := 0; i < len(time); i++ {
		k1 := time[i] % 60
		buf[k1]--
		k := (60 - k1) % 60
		c += buf[k]
	}
	return c
}

func main() {
	fmt.Println("a")
}

File: P/1010-pairs-of-songs-with-total-durations-divisible-by-60/main.rs
struct Solution {}

impl Solution {
  pub fn num_pairs_divisible_by60(time: Vec<i32>) -> i32 {
    let mut buf: [i64; 60] = [0; 60];

    time.iter().for_each(|&v| {
      buf[(v % 60) as usize] += 1;
    });

    let mut res: i64 = 0;
    (1..30).for_each(|v| {
      res += buf[v] * (buf[60 - v]);
    });
    (res + buf[0] * (buf[0] - 1) / 2 + buf[30] * (buf[30] - 1) / 2) as i32
  }
}
